test/framework/cloudstack.go

```diff
@@ -122,25 +131,27 @@ func CloudStackCredentialsAz1() string {
 
 func NewCloudStack(t *testing.T, opts ...CloudStackOpt) *CloudStack {
 	checkRequiredEnvVars(t, requiredCloudStackEnvVars)
+	cmk := buildCmk(t)
 	c := &CloudStack{
-		t: t,
+		t:         t,
+		cmkClient: cmk,
 		fillers: []api.CloudStackFiller{
 			api.RemoveCloudStackAzs(),
 			api.WithCloudStackAzFromEnvVars(cloudstackAccountVar, cloudstackDomainVar, cloudstackZoneVar, cloudstackCredentialsVar, cloudstackNetworkVar,
 				cloudstackManagementServerVar, api.WithCloudStackAz),
 			api.WithCloudStackStringFromEnvVar(cloudstackSshAuthorizedKeyVar, api.WithCloudStackSSHAuthorizedKey),
 			api.WithCloudStackStringFromEnvVar(cloudstackComputeOfferingLargeVar, api.WithCloudStackComputeOfferingForAllMachines),
 		},
+		templatesRegistry: &templateRegistry{cache: map[string]string{}},
```

Comment:
super nit: can we move this down so the initialization of the templatesRegistry happens all at once?